lib/stream.js

```diff
@@ -463,7 +495,7 @@ Stream.prototype.error = function (error) {
     error = new Error(format.apply(null, arguments));
 
   debug(error.message);
-  this.emit('error', error);
+  this.socket.close();
```

Comment:
This is only a temporary solution. Need to investigate how to reliably trigger `closed` function when an error occurs.